pkg/owner/manager.go

```diff
@@ -352,31 +352,35 @@ func getOwnerInfo(ctx, logCtx context.Context, etcdCli *clientv3.Client, ownerPa
 	}
 	if err != nil {
 		logutil.Logger(logCtx).Warn("etcd-cli get owner info failed", zap.Error(err))
-		return "", nil, op, 0, errors.Trace(err)
+		return "", nil, op, 0, 0, errors.Trace(err)
 	}
 	if len(resp.Kvs) == 0 {
-		return "", nil, op, 0, concurrency.ErrElectionNoLeader
+		return "", nil, op, 0, 0, concurrency.ErrElectionNoLeader
 	}
 
 	var ownerID []byte
 	ownerID, op = splitOwnerValues(resp.Kvs[0].Value)
 	logutil.Logger(logCtx).Info("get owner", zap.ByteString("owner key", resp.Kvs[0].Key),
 		zap.ByteString("ownerID", ownerID), zap.Stringer("op", op))
-	return string(resp.Kvs[0].Key), ownerID, op, resp.Kvs[0].ModRevision, nil
+	return string(resp.Kvs[0].Key), ownerID, op, resp.Kvs[0].CreateRevision, resp.Kvs[0].ModRevision, nil
```

Comment:
we only need watch from current revision + 1, see this for an example https://github.com/pingcap/tidb/blob/87244edabf0ebae9870411cb83d91a9add4a797e/pkg/ddl/schemaver/syncer.go#L469-L473

Reply:
Modified